chapter2/tests/t19/terraform/_modules/container/variables.tf

```diff
@@ -0,0 +1,64 @@
+variable "cont" {
+  description = "Name of the container"
+  type        = string
+}
+
+variable "container_image" {
+  description = "Value of the name for the Docker container"
+  type        = string
+}
+
+variable "container_image_keep_locally" {
+  description = "If true, then the Docker image won't be deleted on destroy operation. If this is false, it will delete the image from the docker local storage on destroy operation."
+  type        = bool
+  default     = false
+}
+
+variable "container_name" {
+  description = "Value of the name for the Docker container"
+  type        = string
+  validation {
+    condition     = can(regex("^(saritasa-devops-camps-2023-).*", var.container_name))
+    error_message = "Container name should be prefixed with saritasa-devops-camps-2023-"
+  }
```

Comment:
module should be always generic. Such validation should be at the root level variables.tf

Reply:
> module should be always generic. Such validation should be at the root level variables.tf

@dmitry-mightydevops
Done
I moved it to the base layer, but there is an unpleasant moment with the fact that validation is now specified separately for each container. I haven't been able to make it more elegant yet.